designcompose/src/main/java/com/android/designcompose/squoosh/SquooshRender.kt

```diff
@@ -111,6 +113,16 @@ internal fun Modifier.squooshRender(
                             parentVariableState
                         }
                     val computedLayout = node.computedLayout ?: return
+                    // Translate by the scroll amount if the parent node is the scroll view
+                    val scroll = node.parent?.view == parentNode
+                    if (scroll) {
+                        drawContext.canvas.save()
+                        drawContext.canvas.translate(scrollOffset.value.x, scrollOffset.value.y)
+                    }
+                    // Call scrollRestore() whenever returning from renderNode() in order to restore
+                    // the canvas matrix/clip state. This is a good candidate for RAII but this is
+                    // not supported by Kotlin.
+                    val scrollRestore = { if (scroll) drawContext.canvas.restore() }
```

Comment:
Do you know if there's a cost to making a closure in Kotlin? We'll make this for every node, not just those that scroll, so it might be better to just duplicate the if statement below (I agree there must be a nicer way to structure this than this kind of manual counting... but I don't know what it is either).

Reply:
Good point, it looks like there is a small cost to capturing all the local variables. I couldn't find a way to do local inline functions, so I'll copy the code and add a comment about it.